cpp/splinepy/py/py_spline.hpp

```diff
@@ -814,11 +844,43 @@ inline py::list ExtractBezierPatches(const PySpline& spline) {
 }
 
 /// boundary spline extraction
-inline PySpline ExtractBoundary(const PySpline& spline,
-                                const int& boundary_normal_axis,
-                                const int& extrema) {
-  return PySpline(
-      spline.Core()->SplinepyExtractBoundary(boundary_normal_axis, extrema));
+inline py::list ExtractBoundaries(const PySpline& spline,
+                                  const py::array_t<int>& boundary_ids) {
+  // Init return value
+  py::list boundary_splines{};
+  const int n_boundaries = boundary_ids.size();
+  int* bid_ptr = static_cast<int*>(boundary_ids.request().ptr);
+  if (boundary_ids.size() == 0) {
+    for (int i{}; i < spline.para_dim_ * 2; ++i) {
+      boundary_splines.append(
+          PySpline(spline.Core()->SplinepyExtractBoundary(i)));
+    }
+  } else {
+    const int max_bid = spline.para_dim_ * 2 - 1;
+    for (int i{}; i < n_boundaries; ++i) {
+      const int& bid = bid_ptr[i];
+      if (bid < 0 || bid > max_bid) {
+        splinepy::utils::PrintAndThrowError("Requested Boundary ID :",
+                                            bid,
+                                            "exceeds admissible range.");
+      }
+      boundary_splines.append(
+          PySpline(spline.Core()->SplinepyExtractBoundary(bid_ptr[i])));
+    }
+  }
+
+  return boundary_splines;
+}
+
+/// boundary spline extraction from axis and extreme
+inline PySpline ExtractBoundaryFromAxisAndExtrema(const PySpline& spline,
+                                                  const int& axis,
+                                                  const int& extreme) {
+  // Determine corresponding ID
+  const int boundary_id = (extreme > 0) ? 2 * axis + 1 : 2 * axis;
+
+  // Extract boundary
+  return PySpline(spline.Core()->SplinepyExtractBoundary(boundary_id));
```

Comment:
Thanks for keeping original format. I think one extract function as member func is enough. Could you just add this as helper func?

Reply:
Added the function in `cpp/splinepy/splines/helper/extract.hpp`